MAVProxy/modules/mavproxy_chat/chat_voice_to_text.py

```diff
@@ -55,15 +66,25 @@ def record_audio(self):
 
         # calculate time recording should stop
         curr_time = time.time()
-        time_stop = curr_time + 5
+        time_stop = curr_time + 3
 
         # record until specified time
         frames = []
-        while curr_time < time_stop:
+
+        # logic for recording sound until someone is speaking.
+        isSpeaking = True
+        while curr_time < time_stop or isSpeaking:
             data = stream.read(1024)
             frames.append(data)
+            rms1 = rms(data)
+            if rms1!=0.0:
+                decibel = 20 * math.log10(rms1)
+                isSpeaking = decibel>-80.0  # -80 is the hardcoded threshold. higher number means louder. Set threshold in the range (-100,0)
```

Comment:
If possible let's move this -80 to be a definition at the top of the file where it will be easier to find.